flink-kubernetes/src/main/java/org/apache/flink/kubernetes/configuration/KubernetesConfigOptions.java

```diff
@@ -222,6 +223,25 @@
 		Never
 	}
 
+	/**
+	 * The deployment target names for Kubernetes.
```

Comment:
I would move this to a separate class like the `KubernetesDeploymentTarget` with some additional functionality. I think having all the options somewhere can help later if we want to generate help messages. The class could look like the `YarnDeploymentTarget` and also make the related chagnes in the:

* `KubernetesSessionClusterExecutor`
* `KubernetesClusterClientFactory`
* `KubernetesClusterDescriptor`

Reply:
I have a branch with the proposed change here https://github.com/kl0u/flink/commit/c575102f3c3659885a7176def196666bd046e50f :)

Reply:
I will integrate your commits to keep consistent with Yarn.